### DISCUSSION TITLE
Having a common MapView and BottomSheet

### DISCUSSION THREAD

**Discussion by erwanlpfr** [expo/router#470] [Q&A]
Hello,

I am stuck and would like ideas to implement an efficient pattern of : 

- In the whole app, having a MapView (here's Mapbox). Only one, and make it available for the rest of the app.
- I am using gorhom bottom sheet (I love it) and I would like to have a kind of router to choose from the URL the good view to display : `/services/[...sheet].tsx` for example and still having a bottonsheet in `/index.tsx`.
- And keep the possibility to have a stack of a normal screen to display a "true" page.

To do so, I tried to write my own navigator. But I wasn't convinced, and I got the type of error : route not handled at some point.
Because I went more and more complex, I stopped it and took it back.
So now, I am thinking about having : `/_layout.tsx` with a `<Stack />`. And in `/services/_layout.tsx` having the Mapview and the Modal component which wraps `<Slot />`.
But here, I found the limit of this pattern : not a common Map for the whole app and, having a Slot inside a `<BottomSheet />` do not work always.

So, I am here to ask you if you have any ideas how to implement such a routing system.
For the most of the states, I'll use of course navigation state for most common info, ID but for the map, it'll be a redux store.

Thanks !
Erwan

**Comment by Marcados** [comment]
Hi @erwanlpfr, did you find a solution that suited you ?

**Comment by asharrahman02** [comment]
@erwanlpfr any progress?
